src/install/dracut-install.c

```diff
@@ -625,7 +625,12 @@ static int resolve_deps(const char *src)
         }
 
         fclose(fptr);
-        wait(&err);
+        int err = -1;
+        while (waitpid(ldd_pid, &err, 0) == -1 && errno == EINTR)
+                ;
+        ret = WIFSIGNALED(err) ? 128 + WTERMSIG(err) : WEXITSTATUS(err);
```

Comment:
Different behaviour from b2d3f8610b3d5a3d385164f5459d0c3938831d11. It should not always overwrite possible non-zero `ret` from the `strstr()` matches above.

Reply:
Fixed